app/data_mapper_setup.rb

```diff
@@ -0,0 +1,28 @@
+require 'data_mapper' #= the gem
+require 'dm-validations'
+
+env = ENV['RACK_ENV'] || 'development'
+# checks for current environment from operating system.
+# if RSPEC is running it will set the env to test.
+
+DataMapper.setup(:default, ENV['DATABASE_URL'] || "postgres://localhost/bookmark_manager_#{env}")
```

Comment:
Still failing CI as you're telling postgres to connect to bookmark_manager_test/development. This is Chitter!